src/input/type_record.cc

```diff
@@ -442,8 +445,6 @@ Record &Record::declare_key(const string &key, const KeyType &type,
         data_->declare_key(key, boost::shared_ptr<const TypeBase>(), &type, default_value, description);
     } else {
         // for Array, Double, Integer, we assume no static variables
-    	if (const_cast<KeyType *>( &type )->finish()) check_key_default_value(default_value, type, key);
```

Comment:
Checking default value here produce much better error message, since one can find exact reason for the error from the stack trace. Is it necessary to finish type first? Isn't close enough? Are there some principal reasons why the check can not be done here?